api/authentication.py

```diff
@@ -0,0 +1,67 @@
+from datetime import datetime
+import json
+import requests
+
+from django.conf import settings
+from django.core.cache import cache
+
+from allauth.socialaccount.models import SocialAccount
+from rest_framework.authentication import BaseAuthentication
+
+
+def get_cache_key(token):
+    return f'fxa_token_{token}'
+
+
+class FxaTokenAuthentication(BaseAuthentication):
+    def authenticate(self, request):
+        if 'Authorization' not in request.headers:
+            return None
+
+        authorization = request.headers.get('Authorization')
+        if not authorization.startswith('Bearer '):
+            return None
```

Comment:
I'm doing `return None` on purpose, so that DRF will try to authenticate the request with one of the other authentication backends. (per [docs for custom auth backends](https://www.django-rest-framework.org/api-guide/authentication/#custom-authentication).)